x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/TokenServiceTests.java

```diff
@@ -576,34 +594,40 @@ public void testIndexNotAvailable() throws Exception {
             return Void.TYPE;
         }).when(client).get(any(GetRequest.class), any(ActionListener.class));
 
+        final SecurityIndexManager tokensIndex;
+        if (oldNode != null) {
+            tokensIndex = securityMainIndex;
```

Comment:
worth making both indices unavailable in this case?

Reply:
I believe it is superior to only _mock_ the index manager that is expected to be utilized.
This would catch bugs when the `TokenService` goes for the other index manager than the test expects it to.

Reply:
> This would catch bugs when the TokenService goes for the other index manager than the test expects it to.

That was what I was going after too, mocking both as unavailable, to catch potential bugs

Reply:
Is this what you're proposing?
```
        if (oldNode != null) {
            tokensIndex = securityMainIndex;
            when(securityTokensIndex.isAvailable()).thenReturn(false);
            when(securityTokensIndex.indexExists()).thenReturn(false);
        } else {
            tokensIndex = securityTokensIndex;
            when(securityMainIndex.isAvailable()).thenReturn(false);
            when(securityMainIndex.indexExists()).thenReturn(false);
        }
```

Reply:
yes something to that effect. Maybe my comment was misplaced, how about mocking both as unavailable / nonexistent in https://github.com/elastic/elasticsearch/pull/40742/files/db4d15ccafec6f232e4d7a920ffcc5653f01aabd#diff-61d76d4a779c6ff0040541c1ad000f1eR608

Reply:
Why's that? I believe it's okay to have the _other_ index as _available_ . If the code was buggy and it went for the wrong index the test would fail `assertNull(future.get())` , right?

Reply:
I don't know. Having both as unavailable seemed closer to the original goal of this test "what happens if I try to login and the security index is unavailable?". You have a point that if we mark both as unavailable and future.get() is null, we don't necessarily know if this is because the implementation picked the correct index to search, or because, well, both indices are unavailable. But isn't this what we try to test here ? Thinking out loud, feel free to do eitherway